Storage/tests/Unit/Connection/RestTest.php

```diff
@@ -540,6 +545,108 @@ public function retryFunctionReturnValues()
         ];
     }
 
+    public function retryStrategyCases()
+    {
+        return [
+            // Idempotent operation with retriable error code
+            // and never retry strategy => no retry
+            [
```

Comment:
Perhaps, instead of hardcoding values from another data provider, we could loop through the array returned by the function and modify the `$args` and `$expected`?

Reply:
I did thought about that but decided to stick to this because this made cases more descriptive to anyone who revisits the code.

Should I do the looping?

Reply:
I think that'll be better, if we change the func arguments or the dataprovider values. It makes sure we don't have to change anything in the second data provider.

Possibly we can have 3 cases for every array row.
- Always -> expects retry func to return true
- Never -> Expects a falsy value
- Idempotent -> The first data provdier already covers it, so no need to add this.

Reply:
Okay, so I did this in commit [9b54cab](https://github.com/googleapis/google-cloud-php/pull/5731/commits/9b54cab915f582be22b67bc297da396ce416a418) considering that we never retry non retriable error codes. 

There will be only a very small change to accomodate logic(if it gets confirmed that we need to retry non retriable codes also incase of always retry strategy), so that would not take time.